### PR TITLE
3D RK4

### PR DESCRIPTION
**Pull request by matt-frey**

### CODE REVIEW

File: src/3d/stepper/rk4_utils.f90 [hunk 1]
```diff
@@ -15,20 +15,52 @@ module rk4_utils
...
+
+            ! dB13/dt =
+            Bout(3) = S(7) * Bin(1) & !   dw/dx * B11
+                    + S(8) * Bin(2) & ! + dw/dy * B12
+                    - S(5) * Bin(3) & ! - dv/dy * B13
+                    + S(2) * Bin(5) & ! + du/dy * B23
+                    + S(3) * B33      ! + du/dz * B33
+
+            ! dB22/dt = 2 * (dv/dx * B12 + dv/dy * B22 + dv/dz * B23)
+            Bout(4) = two * (S(4) * Bin(3) + S(5) * Bin(4) + S(6) * Bin(5))
```
Comment (sjboeing) [inline src/3d/stepper/rk4_utils.f90:55]: Isn't B12 the second element ?
Bout(4) = two * (S(4) * Bin(2) + S(5) * Bin(4) + S(6) * Bin(5))

Comment (matt-frey) [reply src/3d/stepper/rk4_utils.f90:55]: Yes, you are right! I did a mistake here.

File: src/3d/stepper/rk4_utils.f90 [hunk 2]
```diff
@@ -15,20 +15,52 @@ module rk4_utils
...
+            ! dw/dx = S(7)
+            ! dw/dy = S(8)
+            ! dw/dz = S(9)
+
+            B33 = get_B33(Bin, volume)
+
+            ! dB11/dt = 2 * (du/dx * B11 + du/dy * B12 + du/dz * B13)
+            Bout(1) = two * (S(1) * Bin(1) + S(2) * Bin(2) + S(3) * Bin(3))
+
+            ! dB12/dt =
```
Comment (sjboeing) [inline src/3d/stepper/rk4_utils.f90:40]: I think you are applying an optimisation here to go to 5 terms (where one has a negative sign), can you send me the notes for reference to help me check (I probably have them somewhere, but it is easier if you can send them)

Comment (matt-frey) [reply src/3d/stepper/rk4_utils.f90:40]: I have them only hand-written. I'll make a photo.

Comment (sjboeing) [reply src/3d/stepper/rk4_utils.f90:40]: Thanks, sounds great!

File: src/3d/stepper/rk4_utils.f90 (hunk 1 above)
Comment (matt-frey) [reply src/3d/stepper/rk4_utils.f90]: Fixed.

**Review by sjboeing — APPROVED** [review]
Other equations look right to me!

**Comment by matt-frey** [comment]
Great! Thanks for reviewing!

**Pull request merged**